### ISSUE TITLE
Packaging difference conda-forge's and pytorch distribution of pytorch

### ISSUE THREAD

**Issue by jjerphan** [conda-forge/pytorch-cpu-feedstock#218]
- What is the plus-value of installing conda-forge's builds of pytorch instead of pytorch's?
 - What was the original reason behind redistributing pytorch on conda-forge?
 - Do you know have a summary of the differences between conda-forge's and the original distribution of pytorch?

**Comment by hmaarrfk** [comment]
> What is the plus-value of installing conda-forge's builds of pytorch instead of pytorch's?

If your computation stack depends on some optimized C binaries and they aren't available on pypi, then conda allows you to have your own "channel" and build in accordance to your requirements.

pytorch's channel depends on anaconda's "default" channel optimizations.

on conda-forge, we depend on conda-forge's optimizations.

On conda-forge, it is a community lead development so you and others can have an impact on the development. For example, I believe we were the first to have Apple silicon (arm) packages.

> What was the original reason behind redistributing pytorch on conda-forge?

The above.

> Do you know a summary of the differences between conda-forge's and the original distribution of pytorch?

- arm compatibility on linux
- lack of windows compatibility (help appreciated)
- usage of shared libprotobuf
- Probably a few more.

**Comment by jjerphan** [comment]
Thank you for those elements.

`pytorch` and `torchvision` (and a few other packages from NVIDIA it seems) are distributed on conda-forge, but the rest of the packages are distributed on the `pytorch` and `nvidia` channels.

In the case of `pytorch`, to which extends is it possible to mix packages coming from different channels? Installing as many packages as possible from `conda-forge` and then from `pytorch` is possible, but would such an installation be prone to problems?

**Comment by hmaarrfk** [comment]
yes prone to problems if you mix conda-forge + defaults

**Comment by jeongseok-meta** [comment]
FWIW, the ['pytorch' channel will no longer be maintained](https://github.com/pytorch/pytorch/issues/138506) (thanks to @hmaarrfk for pointing this out earlier), with PyTorch 2.5 being the last supported version; users are encouraged to switch to 'conda-forge' for future releases.

Additionally, `libtorch` in 'pytorch' was built with `_GLIBCXX_USE_CXX11_ABI=0`, which is incompatible with packages built with `_GLIBCXX_USE_CXX11_ABI=1` (https://github.com/pytorch/pytorch/issues/51039).

**Comment by Tobias-Fischer** [comment]
Should we close here seeing that upstream stopped providing conda packages?

**Comment by hmaarrfk** [comment]
Let’s leave it open for an other year while things settle

**Comment by OverLordGoldDragon** [comment]
**TL;DR** how to install `pytorch=2.6.0` with CUDA 12.6?

If it's the right place to ask: previously (on official [get started](https://pytorch.org/get-started/locally/)) there was e.g. 

```
conda install pytorch==2.5.1 torchvision==0.20.1 torchaudio==2.5.1 pytorch-cuda=12.4 -c pytorch -c nvidia
```

What's the equivalent of that for `-c conda-forge` moving forward, with a target CUDA version? If not equivalent, what's recommended best practice?

**Comment by OverLordGoldDragon** [comment]
Guess it's "too TLDR", the CUDA 12.6 part is important. Opened new [issue](https://github.com/conda-forge/pytorch-cpu-feedstock/issues/381), linking here for reference.
